File: Modules/SEDO/EMS/SIE.Web.EMS.Purchases/FixtureReceives/Commands/SaveFixtureReceiveCommand.cs
using SIE.EMS.Purchases.FixtureReceives;
using SIE.Web.Command;

namespace SIE.Web.EMS.Purchases.FixtureReceives.Commands
{
    /// <summary>
    /// 保存设备接收
    /// </summary>
    [JsCommand("SIE.Web.EMS.Purchases.FixtureReceives.Commands.SaveFixtureReceiveCommand")]
    public class SaveFixtureReceiveCommand : FormSaveCommand
    {
        /// <summary>
        /// 执行命令
        /// </summary>
        /// <param name="args">参数</param>
        /// <param name="scope">作用域</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var list = GetDeserializeData(args, scope);
            var entity = list.Count > 0 ? list[0] : null;
            if (entity != null)
            {
                var eq = entity as FixtureReceive;
                RT.Service.Resolve<FixtureReceiveController>().SaveFixtureReceive(eq);
            }
            return entity;
        }
    }
}

File: Modules/SEDO/EMS/SIE.Web.EMS.Purchases/FixtureReceives/Commands/SaveFixtureReceiveCommand.js
SIE.defineCommand('SIE.Web.EMS.Purchases.FixtureReceives.Commands.SaveFixtureReceiveCommand', {
    extend: 'SIE.cmd.FormSave',
    meta: { text: "保存", group: "edit", iconCls: "icon-SaveEntity icon-blue" },
    /**
     * @override 保存后处理
     * @param {any} view
     * @param {any} res
     */
    onSaved: function (view, res) {
        var me = this;
        var current = view.getCurrent();
        current.markSaved();
        me.onSavedMsg(view, res);
        CRT.Workbench.closeCurrentTab();
        CRT.Event.fire("SIE.EMS.Purchases.FixtureReceives.FixtureReceive_refresh");
    },
    execute: function (view, source) {
        var me = this;
        view.getCurrent().dirty = true;
        var childView = view._children.first(function (p) { return p.model === "SIE.EMS.Purchases.FixtureReceives.FixtureReceiveDetail"; });
        if (childView) {
            childView.getData().getData().items.forEach(function (item) {
                item.dirty = true;
            });
        }
        me.doSave(view);
    }
});